src/Catalog/Registration/RegistrationMakerCatalogItem.cs

```diff
@@ -214,5 +215,25 @@ public override IGraph CreatePageContent(CatalogContext context)
                 throw new Exception(string.Format("Exception processing catalog item {0}", _catalogUri), e);
             }
         }
+
+        private string GetIconUrl()
+        {
+            var subject = _catalogItem.CreateUriNode(_catalogUri);
+            var packageId = _catalogItem.GetTriplesWithSubjectPredicate(subject, _catalogItem.CreateUriNode(Schema.Predicates.Id)).FirstOrDefault().Object.ToString();
```

Comment:
I see we fetch the package ID and version like this in a couple places in this file...could we share this code?

Reply:
Updated